### PR TITLE
Add HMR with webpack on server

### PR DESCRIPTION
**Pull request by pauloedurezende**
- Remove nodemon config file
- Add webpack
- Add HMR (#27)
- Update Start, Build and Serve scripts
- Fix Jest (#2)

PS: I've preserved some packages and scripts that use Nodemon, but I plan to remove it later, leaving everything in the webpack

### CODE REVIEW

File: packages/server/src/config.js [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
+import { resolve } from 'path';
 import dotenvSafe from 'dotenv-safe';
 
-const root = path.join.bind(this, __dirname, '../');
-
 dotenvSafe.load({
-  path: root('.env'),
-  sample: root('.env.example'),
+  path: resolve('.env'),
+  sample: resolve('.env.example'),
```
Comment (jgcmarins) [inline packages/server/src/config.js]: nice, but I think this isn't enough

**Comment by pauloedurezende** [comment]
@sibelius Basically I will not be able to tell you about the performance, but I have seen reports that, by packaging the server, you have a time gain at startup, since the webpack performs optimization of the code. I will perform some more tests to analyze it...

**Comment by sibelius** [comment]
I've run `yarn start:server` on root

and I've got 2 problems

```jsx
@entria/server: webpack is watching the files…
@entria/server: (node:57432) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
```

`DeprecationWarning: Tapable.plugin is deprecated`, how can we solve this?

there are a lot of errors as well

image


2 thinks is missing `.mjs` loader and it should also ignore `node_modules` of root

```jsx
{
        test: /\.mjs$/,
        include: /node_modules/,
        type: 'javascript/auto',
      },
```

```jsx
externals: [
    WebpackNodeExternals({
      whitelist: ['webpack/hot/poll?1000'],
    }),
    WebpackNodeExternals({
      modulesDir: path.resolve(__dirname, '../../node_modules'),
    }),
  ],
```

server is not starting yet

**Comment by PlayMa256** [comment]
`DeprecationWarning: Tapable.plugin is deprecated` is related to one of the plugins used, being only compatible with webpack 3, in this case, `ReloadServerPlugin`. The migration to v4 is easy, if the author does not want to update it, we could update that locally or even fork it.

File: packages/server/webpack/webpack.dev.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const ReloadServerPlugin = require('reload-server-webpack-plugin');
+
+const common = require('./webpack.common');
+
+module.exports = merge.smart(common, {
+  mode: 'development',
+  watch: true,
+  stats: 'errors-only',
+  entry: {
+    server: ['webpack/hot/poll?1000'],
```
[the remark this answers is not in the document]
Comment (PlayMa256) [reply packages/server/webpack/webpack.dev.js]: He is refeering to it being at the most top level `node_modules` folder

**Comment by PlayMa256** [comment]
Share the repo with us, we can see what is making tests hang

**Comment by sibelius** [comment]
tks for the work, gonna merge now

and improve it later on, based on this https://github.com/entria/entria-fullstack/pull/30#issuecomment-441205395

if we have more than one webpack it could cause some troubles

**Comment by jgcmarins** [comment]
Great job here

**Pull request merged**